O365DSCReverse: Add missing EXOAcceptedDomain, EXOAntiPhishPolicy, EXOAntiPhishRule

reijoh opened the pull request:
<!--
    Thanks for submitting a Pull Request (PR) to this project.
    Your contribution to this project is greatly appreciated!

    Please make sure you have read the [Contribution Guidelines](https://github.com/PowerShell/SharePointDsc/wiki/Contributing%20to%20SharePointDsc).

    Please prefix the PR title with the resource name,
    e.g. 'ResourceName: My short description'.
    If this is a breaking change, then also prefix the PR title
    with 'BREAKING CHANGE:',
    e.g. 'BREAKING CHANGE: ResourceName: My short description'.

    You may remove this comment block, and the other comment blocks, but please
    keep the headers and the task list.
-->
#### Pull Request (PR) description
<!--
    Replace this comment block with a description of your PR.
-->

#### This Pull Request (PR) fixes the following issues
<!--
    If this PR does not fix an open issue, replace this comment block with None.
    If this PR resolves one or more open issues, replace this comment block with
    a list the issues using a GitHub closing keyword, e.g.:
    - Fixes #123
    - Fixes #124
-->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/microsoft/office365dsc/228)
<!-- Reviewable:end -->

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/Office365DSC?pullRequest=228) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/Office365DSC?pullRequest=228) before we can accept your contribution.<br/><br/>:x: reijoh [sign now](https://cla.opensource.microsoft.com/microsoft/Office365DSC?pullRequest=228)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/Office365DSC?pullRequest=228) it.</sub>

NikCharlebois commented:
@reijoh thanks for the PR. It appears we may have some overlapping changes between:
https://github.com/microsoft/Office365DSC/pull/221

And the PR you just submitted. What I would like to ask of you is that you give the PR link above a try and see if it actually does the Extract of the AntiPhishPolicy, Rule and ATP as you would expect it to. We are in the process of refactoring the logic within our Export-TargetResource so that all the extraction logic is done in there, and that O365DSC.Reverse becomes just a class to collect results from the various resources' methods. Thanks

reijoh commented:
> 
> 
> @reijoh thanks for the PR. It appears we may have some overlapping changes between:
> #221
> 
> And the PR you just submitted. What I would like to ask of you is that you give the PR link above a try and see if it actually does the Extract of the AntiPhishPolicy, Rule and ATP as you would expect it to. We are in the process of refactoring the logic within our Export-TargetResource so that all the extraction logic is done in there, and that O365DSC.Reverse becomes just a class to collect results from the various resources' methods. Thanks

I could not find EXOAcceptedDomain in #221. I have removed EXOAntiPhishPolicy and EXOAntiPhishRule and updated EXOAcceptedDomain logic to what you have done for EXOAntiPhish. I will do some testing tomorrow.

reijoh commented:
I justed tested #221, and can confirm that extraction of chckEXOAntiPhishPolicy works. Also, many components have more parameters now.
The change I did with EXOAcceptedDomain do not work, it asks for Identity and I can see that a change must be made in Export-TargetResource of MSFT_EXOAcceptedDomain.psm1 before it will work. I can do the change if you do not do it yourself, but will wait for #221 to be merged first.

This PR should be rejected.
